### ISSUE TITLE
[Ops] Support for reduce average op (ttnn.mean) 

### ISSUE THREAD

**Issue by nvukobratTT** [tenstorrent/tt-forge-fe#114]

**Comment by nvukobratTT** [comment]
@chandrasekaranpradeep this one is unblocked. Can you confirm if it works? :))

**Comment by chandrasekaranpradeep** [comment]
@nvukobratTT There are two test cases in the [forge/test/mlir/test_ops.py](https://github.com/tenstorrent/tt-forge-fe/blob/main/forge/test/mlir/test_ops.py)
1) [test_reduce_mean](https://github.com/tenstorrent/tt-forge-fe/blob/3817b9b2e559cb97497e035c7cf62e69bac455b3/forge/test/mlir/test_ops.py#L361)
         In this test cases, TTNN: Issues with ttnn implicit reshape is resolved
        The below test cases alone is failing with pcc issue (PCC: 0.7203957195745748)
       `pytest forge/test/mlir/test_ops.py::test_reduce_mean[-1-input_shape2] -vss `
2) [test_mean](https://github.com/tenstorrent/tt-forge-fe/blob/3817b9b2e559cb97497e035c7cf62e69bac455b3/forge/test/mlir/test_ops.py#L416)       
          In this test cases, the FFE: Unsupported squeeze operation issue is resolved, now all the test cases here are failing with `Unable to reshape a tensor in TILE_LAYOUT to non-tile height and width! Please convert the tensor to ROW_MAJOR_LAYOUT first.` issue

**Comment by chandrasekaranpradeep** [comment]
@nvukobratTT Created a PR for reduce average op test cases - https://github.com/tenstorrent/tt-forge-fe/pull/349

**Comment by chandrasekaranpradeep** [comment]
Created a issue for the reduce average (i.e ttnn.mean) data mismatch  
TT-MLIR - https://github.com/tenstorrent/tt-mlir/issues/869
TT-Metal - https://github.com/tenstorrent/tt-metal/issues/13621

Data Mismatch Failing Test cases:
```
Reduce_avg:
    Input_shape = (1, 12, 3200)
    dim = -1
    keep_dim = True
pytest forge/test/mlir/test_ops.py::test_reduce_mean[-1-input_shape2] -vss
```

**Comment by chandrasekaranpradeep** [comment]
> Created a issue for the reduce average (i.e ttnn.mean) data mismatch
> TT-MLIR - https://github.com/tenstorrent/tt-mlir/issues/869
> TT-Metal - https://github.com/tenstorrent/tt-metal/issues/13621

@nvukobratTT  Reduce average op configuration (i.e Input_shape = (1, 12, 3200), dim = -1, keep_dim = True) is present in the Llama 3b model. Can we change the priority of the issue from P1 to P0?

**Comment by chandrasekaranpradeep** [comment]
In tt-metal latest main,  `Unable to reshape a tensor in TILE_LAYOUT to non-tile height and width! Please convert the tensor to ROW_MAJOR_LAYOUT first.` issue is resolved for [test_mean](https://github.com/tenstorrent/tt-forge-fe/blob/3817b9b2e559cb97497e035c7cf62e69bac455b3/forge/test/mlir/test_ops.py#L416) pytest. 
Once the tt-mlir is uplifted with latest tt-metal main, need to verify the issue is resolved in forge or not and update the [test_mean](https://github.com/tenstorrent/tt-forge-fe/blob/3817b9b2e559cb97497e035c7cf62e69bac455b3/forge/test/mlir/test_ops.py#L416)

**Comment by nvukobratTT** [comment]
> In tt-metal latest main, `Unable to reshape a tensor in TILE_LAYOUT to non-tile height and width! Please convert the tensor to ROW_MAJOR_LAYOUT first.` issue is resolved for [test_mean](https://github.com/tenstorrent/tt-forge-fe/blob/39a7977377dc0afef1d22d16f8e586eb234855a1/forge/test/mlir/test_ops.py#L417C1-L417C38) pytest. Once the tt-mlir is uplifted with latest tt-metal main, need to verify the issue is resolved in forge or not and update the [test_mean](https://github.com/tenstorrent/tt-forge-fe/blob/39a7977377dc0afef1d22d16f8e586eb234855a1/forge/test/mlir/test_ops.py#L417C1-L417C38)

Yup! I think uplift of this one is already WIP :)) We expect it to land soon. 

Please keep track of it and test out this op accordingly :))

**Comment by chandrasekaranpradeep** [comment]
> Yup! I think uplift of this one is already WIP :)) We expect it to land soon.
> Please keep track of it and test out this op accordingly :))

Sure Nikola
